File: Engine/src/main/scala/fr/linkit/engine/connection/packet/persistence/serializor/ConstantProtocol.scala
/*
 *  Copyright (c) 2021. Linkit and or its affiliates. All rights reserved.
 *  DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 *  This code is free software; you can only use it for personal uses, studies or documentation.
 *  You can download this source code, and modify it ONLY FOR PERSONAL USE and you
 *  ARE NOT ALLOWED to distribute your MODIFIED VERSION.
 *
 *  Please contact maximebatista18@gmail.com if you need additional information or have any
 *  questions.
 */

package fr.linkit.engine.connection.packet.persistence.serializor

object ConstantProtocol {
    final val Class: Byte = 0
    final val ContextRef: Byte = 1
    final val PoolRef: Byte = 2

    final val String: Byte = 3
    final val Int: Byte = 4
    final val Short: Byte = 5
    final val Long: Byte = 6
    final val Byte: Byte = 7
    final val Double: Byte = 8
    final val Float: Byte = 9
    final val Boolean: Byte = 10
    final val Char: Byte = 11

    final val Array: Byte = 12
    final val Object: Byte = 13

    final val Null: Byte = -128
}

File: Engine/src/main/scala/fr/linkit/engine/connection/packet/persistence/serializor/ConstantProtocol.java
package fr.linkit.engine.connection.packet.persistence.serializor;

public class ConstantProtocol {

    private static byte i = 0;
    public static final byte Class = i++;
    public static final byte SyncClass = i++;
    public static final byte ContextRef = i++; //2

    public static final byte String = i++; //3
    public static final byte Int = i++;
    public static final byte Short = i++;
    public static final byte Long = i++;
    public static final byte Byte = i++; //7
    public static final byte Double = i++;
    public static final byte Float = i++;
    public static final byte Boolean = i++; //10
    public static final byte Char = i++; //11

    public static final byte Enum = i++; //12
    public static final byte Object = i++; //13
    public static final byte Array = i++;

    public static final byte ChunkCount = i;

}
